### ISSUE TITLE
Fluid duplication glitch

### ISSUE THREAD

**Issue by Ephys** [Glitchfiend/BiomesOPlenty#321]
You can still use a bucket to place a fluid as long as the fluid amount is above 0, even if it is under 1000mB.

Also, I'm not sure of why you implemented your bucket using ItemFluidContainer instead of just making a new instance of ItemBucket (and passing your FluidBlock) and using the FluidContainerRegistry because your implementation causes a lot of problems:
- Using the bucket in anything that manages IFluidContainers correctly will result in a BiomesOPlenty:bopBucket instead of a vanilla bucket.
- That bucket cannot pick up any liquid from the ground
- That bucket will accept any fluid using something that manages IFluidContainers (like fluid transposers), but the texture will still be the empty bucket texture and the name will be... weird ?
- There doesn't seem to be a bucket container for the BOP fluids in the FluidContainerRegistry
- the bug reported here (well that can be fixed but it's weird)
- [fixed] They won't work on most tanks as the tanks look through the fluidcontainer registry to know what fluid your bucket is holding.
- (OCD level) Isn't it less efficient to store the fluid type /per/ bucket as NBT instead of storing it per bucket type by making a new ItemBucket instance ? ~

TL;DR: It's not usefull to reimplement the vanilla bucket when all it does is nothing more than being a bucket. You should keep that for new fluid containers (and add support for every fluid while you're at it [or whitelist your fluids]).

Obviously that's just my opinion, feel free to tell me to mind my own business.

**Comment by Adaptivity** [comment]
@Adubbz did you solved those concerns in 1.8.9? I hope you didn't just ported old 1.7 code :) Just asking.

**Comment by Adubbz** [comment]
This should actually be resolved, we used the standard Vanilla bucket in 1.8.9 rather than implementing our own as was suggested here
